packages/header-change-detection/lib/lambda/header-check.ts

```diff
@@ -0,0 +1,300 @@
+import axios from "axios";
+import { DynamoDBClient, BatchGetItemCommand, BatchGetItemCommandInput, KeysAndAttributes, UpdateItemCommandInput, UpdateItemCommand, AttributeValueUpdate } from "@aws-sdk/client-dynamodb";
+import { PublishCommand, PublishInput, SNSClient } from "@aws-sdk/client-sns";
+
+const URLS = process.env.URLS;
+const HEADERS = process.env.HEADERS;
+const TABLE = process.env.TABLE!;
+
+const config = ""
+const DB_CLIENT = new DynamoDBClient(config);
+
+const securityHeaders = HEADERS?.split(",") || []
+
+type Headers = Map<string, string | undefined>
+type URLHeaders = Map<string, Headers>
+
+export const handler = async () => {
+  const urls = URLS?.split(',') || [];
+
+  // Fetch stored headers
+  const [storedUrlHeaders, currentUrlHeaders] = await Promise.all([
+    getStoredValues(urls),
+    fetchHeaders(urls)
+  ]);
+
+  // Find any differences between the headers
+  const headerDifferences = new Map() as Map<string, Difference[]>;
+  let differencesDetected = false;
+  const dbUpdates = urls.map(url => {
+    const currentHeaders = currentUrlHeaders.get(url)
+    const storedHeaders = storedUrlHeaders.get(url) || new Map()
```

Comment:
Does TS infer the default `Map`'s internal type correctly here?
Would it be worth typing the default `Map` here so it's internals are known (if TS isn't able to infer from usage)
eg. `new Map<string, Headers>` or `new UrlHeaders()` if that's possible (not sure if TS will compile that back to the original type? :man_shrugging:

Reply:
You can't just call `new UrlHeaders()` because it's a type. I think if I rewrote it to extend the Map class it would work but that doesn't feel worth it imo.

You are right though, it doesn't know the type and sees it as `Map<any, any>`. I might use the `as` keyword here to make that slightly nicer.